domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1108 @@
+      }
+      else {
+        $domain->saveDefault();
+      }
+    }
+    $this->logger()->info(dt('!domain set to primary domain.',
+      ['!domain' => $domain->getHostname()]));
+  }
+
+  /**
+   * Deactivates the domain.
+   *
+   * @param $domain_id
+   *   The numeric id or hostname of the domain to disable.
+   * @usage drush domain-disable example.com
+   * @usage drush domain-disable 1
+   *
+   * @command domain:disable
+   * @aliases domain-disable
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function disable($domain_id) {
+    // Resolve the domain.
+    if ($domain = $this->getDomainFromArgument($domain_id)) {
+      if ($domain->status()) {
+        $domain->disable();
+        $this->logger()->info(dt('!domain has been disabled.',
+          ['!domain' => $domain->getHostname()]));
+      }
+      else {
+        $this->logger()->info(dt('!domain is already disabled.',
+          ['!domain' => $domain->getHostname()]));
+      }
+    }
+  }
+
+  /**
+   * Activates the domain.
+   *
+   * @param $domain_id
+   *   The numeric id or hostname of the domain to enable.
+   * @usage drush domain-disable example.com
+   * @usage drush domain-enable 1
+   *
+   * @command domain:enable
+   * @aliases domain-enable
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function enable($domain_id) {
+    // Resolve the domain.
+    if ($domain = $this->getDomainFromArgument($domain_id)) {
+      if (!$domain->status()) {
+        $domain->enable();
+        $this->logger()->info(dt('!domain has been enabled.',
+          ['!domain' => $domain->getHostname()]));
+      }
+      else {
+        $this->logger()->info(dt('!domain is already enabled.',
+          ['!domain' => $domain->getHostname()]));
+      }
+    }
+  }
+
+  /**
+   * Changes a domain label.
+   *
+   * @param $domain_id
+   *   The numeric id or hostname of the domain to relabel.
+   * @param $name
+   *   The name to use for the domain.
+   * @usage drush domain-name example.com Foo
+   * @usage drush domain-name 1 Foo
+   *
+   * @command domain:name
+   * @aliases domain-name
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function renameDomain($domain_id, $name) {
+    // Resolve the domain.
+    if ($domain = $this->getDomainFromArgument($domain_id)) {
+      $domain->saveProperty('name', $name);
+    }
+  }
+
+  /**
+   * Changes a domain name.
+   *
+   * @param $domain_id
+   *   The numeric id or hostname of the domain to rename.
+   * @param $machine_name
+   *   The machine-readable name to use for the domain.
+   * @usage drush domain-machine-name example.com foo
+   * @usage drush domain-machine-name 1 foo
+   *
+   * @command domain:machine-name
+   * @aliases domain-machine-name
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function machineName($domain_id, $machine_name) {
```

Comment:
This was a super-admin tool in D7 when we didn't use machine name for data storage, we just used numeric id. The field data now uses machine_name as the value, which makes this command very volatile. 

If we change a machine name, we have to change any field storage reference to the old machine name.

Reply:
ok, might park that one and disable this fn. Sounds like a reassign all over again...